lua/telescope/sorters.lua

```diff
@@ -444,6 +445,14 @@ sorters.get_fzy_sorter = function(opts)
     discard = true,
 
     scoring_function = function(_, prompt, line)
+      local i = prompt:find(":", 1, true) -- Specify the line number by colon
+      if i then
+        global_state.set_global_key("line_no", tonumber(prompt:sub(i + 1)))
+        prompt = prompt:sub(1, i - 1)
+      else
+        global_state.set_global_key("line_no", nil)
+      end
+
```

Comment:
lets not do that in sorter, if we do it here we need to do that in all sorters (also external sorters like fzf-native) and that doesnt scale.

we should rather do that in the main loop, here or in the `_get_next_filtered_prompt` function

https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/pickers.lua#L473-L474